cpp/cmake/cudf-build-config.cmake.in

```diff
@@ -38,9 +56,34 @@ set(CUDF_MIN_VERSION_rmm "${CUDF_VERSION_MAJOR}.${CUDF_VERSION_MINOR}")
 include(@CUDF_SOURCE_DIR@/cmake/thirdparty/CUDF_GetRMM.cmake)
 # find gtest
 include(@CUDF_SOURCE_DIR@/cmake/thirdparty/CUDF_GetGTest.cmake)
+# find GoogleBench
+CPMFindPackage(NAME benchmark
+    VERSION         1.5.2
+    GIT_REPOSITORY  https://github.com/google/benchmark.git
+    GIT_TAG         v1.5.2
+    GIT_SHALLOW     TRUE
+    OPTIONS         "BENCHMARK_ENABLE_TESTING OFF"
+                    "BENCHMARK_ENABLE_INSTALL OFF")
+
+function(add_cudf_library_alias alias target)
+    if(NOT TARGET ${alias} AND (TARGET ${target}))
+        add_library(${alias} ALIAS ${target})
+    endif()
+endfunction()
+
+# Add cudf::* aliase targets like using an installed cuDF
+add_cudf_library_alias(cudf::gtest GTest::gtest)
```

Comment:
Overall I think we need improve the promotion to global logic.

- The `GTest::` targets need to be added to `_possible_targets_to_promote`
- `fix_cmake_global_defaults` needs to be aware of `ALIAS` targets and to set `ALIAS_GLOBAL` to true on those